spec/models/snack_spec.rb

```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+RSpec.describe Snack, type: :model do
+  describe 'validations' do
+    it { should validate_presence_of :name }
+    it { should validate_presence_of :price }
+    it { should have_many :machine_snacks }
+    it { should have_many(:machines).through(:machine_snacks) }
+  end
+
+
+end
```

Comment:
This is not a comment for this file, but rather just a reminder to test your joins table as well. Make sure your joins table has belongs to relationships to snack and to machine.